whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NotificationFileDao.java

```diff
@@ -38,19 +39,28 @@ INSERT INTO notification_file (version_id, created, payload)
         jdbcTemplate.update(sql, file.versionId(), file.created(), file.payload());
     }
 
-    public Optional<NotificationFile> findLastNotification(final NrtmSourceModel source) {
+    public void update(final NotificationFile file) {
+        final String sql = """
+            UPDATE notification_file set version_id =? , created =?,  payload=? 
+            WHERE id =?
+            """;
+        jdbcTemplate.update(sql, file.versionId(), file.created(), file.payload(), file.id());
+    }
+
+    public Optional<NotificationFile> findLastNotification(final NrtmSource source) {
         try {
             final String sql = """
                 SELECT nf.id, nf.version_id, nf.created, nf.payload
                 FROM notification_file nf
                 JOIN version_info vi ON vi.id = nf.version_id
                 WHERE vi.source_id = ?
-                ORDER BY nf.created DESC LIMIT 1
                 """;
             return Optional.ofNullable(jdbcTemplate.queryForObject(sql, rowMapper, source.getId()));
         } catch (final EmptyResultDataAccessException e) {
             return Optional.empty();
+        } catch (final IncorrectResultSizeDataAccessException exception) {
+            LOGGER.error("More than one entry found for a source {} in update notification file", source.getName());
+            throw new IllegalStateException("More than one entry found for a source {} in update notification file");
```

Comment:
Don't you need to parse the {} here? the final detailed message will keep {} instead of the source